spanner/read.go

```diff
@@ -138,10 +142,32 @@ var _ rowIterator = (*RowIterator)(nil)
 // there are no more results. Once Next returns Done, all subsequent calls
 // will return Done.
 func (r *RowIterator) Next() (*Row, error) {
+	mt := r.meterTracerFactory.createBuiltinMetricsTracer(r.ctx)
 	if r.err != nil {
 		return nil, r.err
 	}
-	for len(r.rows) == 0 && r.streamd.next() {
+	// Start new attempt if we don't have one
+	if mt.currOp.currAttempt == nil {
+		mt.currOp.incrementAttemptCount()
+		mt.currOp.currAttempt = &attemptTracer{
+			startTime: time.Now(),
+		}
+	}
+	defer func() {
+		if mt.method != "" {
+			// record the attempt completion
+			if mt.currOp.currAttempt != nil {
+				mt.currOp.currAttempt.setStatus(status.Code(r.err).String())
+				recordAttemptCompletion(&mt)
+			}
+			statusCode, _ := convertToGrpcStatusErr(r.err)
+			mt.currOp.setStatus(statusCode.String())
+			recordOperationCompletion(&mt)
```

Comment:
This seems to indicate that we will record the latency of the last attempt that finished as the operation latency. That means that an operation that is retried will not contain the entire latency, but only the latency of the last attempt. Is that intentional? If so, could we document that in code to clarify?

Reply:
Actually the currOp start time is initalised in first line of Next() and in recordOperationCompletion we capture to total time including multiple attempts latencies

Reply:
Ack. I see now that there is a separate field for the operation start time.